docs/integration-guides/work-generation.md

```diff
@@ -0,0 +1,209 @@
+
+This will trigger continual work generation, so let it run until a sufficient sample size of times are generated (at least 100 instances). Compute the average of these times which is the number of microseconds to generate work.
+
+**Using Nano Work Server**
+
+If using work server, head over to the README at https://github.com/guilhermelawless/blake2b-pow-bench for details for benchmark script execution.
+
+
+### Example benchmarks
+
+Below are work generation benchmarks from a variety of consumer-grade CPUs and GPUs updated as of March 2020.
+
+**!!!! UPDATES REQUIRED !!!!**
+
+| Device | Epoch v1<br /># PoW/sec |  Epoch v2<br /># PoW/sec |
+|--------|-|-|
+| Nvidia Tesla V100 (AWS) | 6.4 | |
+| Nvidia Tesla P100 (Google,Cloud) | 4.9 | |
+| Nvidia Tesla K80 (Google,Cloud) | 1.64 | |
+| AMD RX 470 OC | 1.59 | |
+| Nvidia GTX 1060 3GB | 1.25 | |
+| Intel Core i7 4790K AVX2 | 0.33 | |
+| Intel Core i7 4790K,WebAssembly (Firefox) | 0.14 | |
+| Google Cloud 4 vCores 0.14-0.16 | 0.14-0.16 | |
+| ARM64 server 4 cores (Scaleway) | 0.05-0.07 | |
+
+---
+
+## Work calculation details
+
+### Work equation
+
+The `"work"` field in transactions contains a 64-bit [nonce](https://en.wikipedia.org/wiki/Cryptographic_nonce) found using the [blake2b hash function](https://blake2.net/).  The nonce satisfies the following equations depending on block height:
+
+**Block Height 1**
+
+The first block on an account-chain doesn't have a previous (head) block, so the account public key is used:
+
+$$
+blake2b(\text{nonce} || \text{public_key}) \ge \text{threshold}
+$$
+
+**Block Height 2 and up**
+
+Once an account has an existing block the previous block hash is used for all blocks going forward:
+
+$$
+blake2b(\text{nonce} || \text{prev_block_hash}) \ge \text{threshold}
+$$
+
+### Difficulty thresholds
+
+The mainnet's base difficulty threshold currently `0xffffffc000000000` for all blocks. To be valid the work field must contain a nonce above this value, but the nodes also prioritize the order in which they process transactions based on how far above this threshold the work value is. Due to the prioritization it may be desirable to generate work further above the threshold to guarantee faster processing by network. To assist integrations with managing these work difficulty levels, nodes monitor the trend of difficulty seen on blocks being processed and make that available via the [`active_difficulty`](/commands/rpc-protocol/#active_difficulty) RPC.
+
+**Development node wallet behavior**
+
+The developer wallet included with the node is configured to pre-cache work as the base threshold and monitor any blocks it publishes to the network for confirmation. If they are not confirmed within 5 seconds, the difficulty on that block will be compared against the active difficulty seen on the network. If the block has a lower work value than the network then new work generation is requested at the higher level.
```

Comment:
We can also mention there is a config for the maximum multiplier (again, we'll have to decide which difficulty that will be based off of)

Reply:
We do use it outside of local work generation, any requests to peers should never be above that maximum multiplier (more concretely, the equivalent difficulty). This is to prevent issues with the node wallet trying to generate work at higher difficulty but this never being completed.

Reply:
Ok, I think I updated this right and included as a notable config callout linked in the update.

Reply:
Also of note for external integrations (not only the dev wallet) since the RPC will reject requests for difficulty levels above this maximum.

Reply:
Moved to a tip callout box below external integrations